kernel/src/process.rs

```diff
@@ -1488,16 +1498,36 @@ impl<C: Chip> ProcessType for Process<'_, C> {
             let _ = writer.write_fmt(format_args!("{}", config));
         });
 
-        let sram_start = self.memory.as_ptr() as usize;
-        let flash_start = self.flash.as_ptr() as usize;
-        let flash_init_fn = flash_start + self.header.get_init_function_offset() as usize;
+        // Print a helpful message on how to re-compile a process to view the
+        // listing file. If a process is PIC, then we also need to print the
+        // actual addresses the process executed at so that the .lst file can be
+        // generated for those addresses. If the process was already compiled
+        // for a fixed address, then just generating a .lst file is fine.
 
-        let _ = writer.write_fmt(format_args!(
-            "\
-             \r\nTo debug, run `make debug RAM_START={:#x} FLASH_INIT={:#x}`\
-             \r\nin the app's folder and open the .lst file.\r\n\r\n",
-            sram_start, flash_init_fn
-        ));
+        self.debug.map(|debug| {
+            if debug.fixed_address_flash.is_some() {
+                // Fixed addresses, can just run `make lst`.
+                let _ = writer.write_fmt(format_args!(
+                    "\
+                     \r\nTo debug, run `make lst` in the app's folder\
+                     \r\nand open the arch.{:#x}.{:#x}.lst file.\r\n\r\n",
+                    debug.fixed_address_flash.unwrap_or(0),
+                    debug.fixed_address_ram.unwrap_or(0)
+                ));
+            } else {
+                // PIC, need to specify the addresses.
+                let sram_start = self.memory.as_ptr() as usize;
+                let flash_start = self.flash.as_ptr() as usize;
+                let flash_init_fn = flash_start + self.header.get_init_function_offset() as usize;
+
+                let _ = writer.write_fmt(format_args!(
+                    "\
+                     \r\nTo debug, run `make debug RAM_START={:#x} FLASH_INIT={:#x}`\
+                     \r\nin the app's folder and open the .lst file.\r\n\r\n",
+                    sram_start, flash_init_fn
```

Comment:
Won't this have an effect on code size?

Reply:
I guess so? Is that bad?

Reply:
There are a lot of strings here, I'm worried this will be a big increase.

Do you have a diff of the size?

Reply:
```
hail: flash use +320 bytes (+0.30%), RAM use +0 bytes(+0.00%)
diff --git a/previous-benchmark-hail b/current-benchmark-hail
index 31658c6..52f9757 100644
--- a/previous-benchmark-hail
+++ b/current-benchmark-hail
@@ -1,6 +1,6 @@
 Tock memory usage report for ./target/thumbv7em-none-eabi/release/hail.elf
-Kernel occupies 106716 bytes of flash
-  103520	code and constant strings
+Kernel occupies 107036 bytes of flash
+  103840	code and constant strings
     3196	variable initializers
 Kernel occupies 16380 bytes of RAM
```

```
nrf52840dk: flash use +240 bytes (+0.19%), RAM use +0 bytes(+0.00%)
diff --git a/previous-benchmark-nrf52840dk b/current-benchmark-nrf52840dk
index c2cc239..ef5eb1a 100644
--- a/previous-benchmark-nrf52840dk
+++ b/current-benchmark-nrf52840dk
@@ -1,6 +1,6 @@
 Tock memory usage report for ./target/thumbv7em-none-eabi/release/nrf52840dk.elf
-Kernel occupies 123668 bytes of flash
-  121888	code and constant strings
+Kernel occupies 123908 bytes of flash
+  122128	code and constant strings
     1780	variable initializers
 Kernel occupies 16384 bytes of RAM
```

From: https://github.com/tock/tock/pull/1930/checks?check_run_id=850230327

Reply:
Ah, I forgot we run size diffs on PRs.

Ok, that seems pretty small so that's fine.